docs/reference-guides/block-api/block-attributes.md

```diff
@@ -167,6 +167,21 @@ _Example_: Extract `src` and `alt` from each image element in the block's markup
 // }
 ```
 
+## Unique Attributes
+
+The `unique` key is used to mark block attributes that should not be copied when a block is duplicated. By default all attribute values will be cloned to the duplicate block. If an attribute sets the `unique` key to `true`, its value will not be copied, and it will instead fall back to the default value if one is supplied.
+
+_Example_: A `resourceId` attribute is marked as `unique` so that it will not be copied to duplicate blocks.
+
+```js
+{
+	resourceId: {
+		type: 'string',
+		unique: true
```

Comment:
I'm wondering if we can just create a new type called `ID` (Borrowed from [graphql](https://graphql.org/learn/schema/#scalar-types)). By definition, it should be a serializable, unique, non-human-readable string. Users can transform it as a number if they wish, but I think string is a good default.

I don't understand how default value could be useful in this case though 🤔. Ideally, **unique** should mean that the value should be unique across the app, and default values break that rule. I understand sometimes it's necessary in the current architecture though, like when pressing `Enter` in a block to create two new blocks while destroying the original one.

Reply:
I think the current types are taken from JSON schema.
- https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/#type-validation
- https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.1.1

It might be ok to deviate from that, I don't know whether there are any definite type restrictions in other parts of the codebase or the REST API. Certainly what's in the blocks package is pretty easy to change.

Reply:
My initial idea was to do something like this. I abandoned the approach for the same reason @talldan pointed out, and because making an `id` type creates much stronger expectations of enforcing uniqueness.

When I raised the issue, for my purposes it was sufficient to make attributes _non-duplicable_ rather than strictly unique, an easier bar to clear. It's pretty clear that the naming in the current implementation conflates these ideas, so I've renamed the attribute key to `duplicable`.